File: C++/Leetcode 1626.cpp
#include <stdio.h>
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int bestTeamScore(vector<int>& scores, vector<int>& ages) {
        vector<pair<int,int>> vect;

        for (int i = 0; i < scores.size(); i++)
        {
            vect.push_back(make_pair(ages[i],scores[i]));
        }

        sort(vect.begin(),vect.end());

        vector<int> DP(scores.size(),0);

        int size = scores.size();
        for (int i = 0; i < size; i++)
        {
            int answer = 0;
            int maxi = vect[i].second;
            for (int j = i - 1; j >= 0; j--)
            {
                if (vect[j].second <= maxi)
                {
                    answer = max(answer,DP[j]);
                }
            }
            DP[i] = maxi + answer;
        }
        int answer = -1;
        for (int i = 0; i < size; i++)
        {
            answer = max(answer, DP[i]);
        }
        return answer;
    }
};

File: Python/Leetcode 1626.py
from ast import List


class Solution:
    def bestTeamScore(self, scores: List[int], ages: List[int]) -> int:
        temp_list = []
        DP = []
        for i in range(0,len(scores)):
            temp_list.append(list([ages[i],scores[i]]))
            DP.append(0)

        temp_list.sort()
        global_answer = 0
        for i in range(0,len(scores)):

            max_score = temp_list[i][1]
            answer = 0

            for j in range(0,i):
                if (temp_list[j][1] <= max_score):
                    answer = max(answer,DP[j])

            DP[i] = max_score + answer
            global_answer = max(global_answer,DP[i])
        
        #print(temp_list)
        #print(DP)
        return global_answer 